### PR TITLE
Remove exact aggregator for histogram instruments

### LINKED ISSUE #2346: Remove the `sdk/metric/aggregator/exact` aggregator
### Problem Statement

This aggregator dates from very early in the prototype. It was used to demonstrate a Summary-like metric aggregation, including the raw data used to calculate precise quantiles. However, it is a misleading type and produces confusion, particularly because the OTLP exporter turns the type into raw gauge points, meaning it exposes a route from Histogram instruments to Gauge data. 

Twice today I answered questions from confused users who used the simple `NewWithExactDistribution()` aggregator selector and got a bunch of gauge points. 

### Proposed Solution

Remove this code.

#### Alternatives

Users can implement this aggregator easily. One of the "near misses" for this code is to support passing raw data through the OTel API directly to another metrics library. This aggregator doesn't support that use-case very well.

### PR DESCRIPTION
**Pull request by jmacd**
This aggregator causes confusing behavior, especially when paired with the OTLP exporter behavior. This aggregator and the corresponding AggregatorSelector (`selector/simple.NewWithExactDistribution()`) gave users a misleading route from histogram instruments to gauge measurements.

Fixes #2346.

### CODE REVIEW

File: bridge/opencensus/aggregation.go [hunk 1]
```diff
@@ -31,127 +31,111 @@ var (
...
 }
 
-var _ aggregation.Aggregation = &ocExactAggregator{}
-var _ aggregation.LastValue = &ocExactAggregator{}
-var _ aggregation.Points = &ocExactAggregator{}
+var _ aggregation.Aggregation = &ocRawAggregator{}
+var _ aggregation.LastValue = &ocRawAggregator{}
 
-// newExactAggregator creates an OpenTelemetry aggreation from OpenCensus points.
+// recordGaugePoints creates an OpenTelemetry aggreation from OpenCensus points.
```
Comment (evantorrie) [inline bridge/opencensus/aggregation.go:61]:
```suggestion
// recordGaugePoints creates an OpenTelemetry aggregation from OpenCensus points.
```

File: bridge/opencensus/aggregation.go [hunk 2]
```diff
@@ -31,127 +31,111 @@ var (
...
-	return t
+func (o *ocRawAggregator) LastValue() (number.Number, time.Time, error) {
+	return o.value, o.time, nil
 }
 
 var _ aggregation.Aggregation = &ocDistAggregator{}
 var _ aggregation.Histogram = &ocDistAggregator{}
 
-// newDistributionAggregator creates an OpenTelemetry aggreation from
+// recordDistributionPoint creates an OpenTelemetry aggreation from
```
Comment (evantorrie) [inline bridge/opencensus/aggregation.go:105]:
```suggestion
// recordDistributionPoint creates an OpenTelemetry aggregation from
```

File: bridge/opencensus/exporter.go [hunk 3]
```diff
@@ -95,18 +96,20 @@ func (d *metricReader) ForEach(_ aggregation.TemporalitySelector, f func(export.
...
+						&descriptor,
+						&ls,
+						agg,
+						ts.StartTime,
+						end,
+					))
+				})
+			fmt.Println("LOOK", err)
+			if err != nil && !errors.Is(err, aggregation.ErrNoData) {
+				fmt.Println("WOMP WOMP")
```
Comment (evantorrie) [inline bridge/opencensus/exporter.go:112]: is `WOMP WOMP` the customary output in this case?

**Review by evantorrie — APPROVED** [review]
Looks good to me. I'm not an expert (nor even conversant) on OpenCensus, so it may be useful to have someone who understands OpenCensus more to take a look at the changes there.

**Comment by dashpole** [comment]
Just to confirm my understanding, since the bridge is now a LastValue aggregation, the opencensus bridge might skip points if the opencensus ReportingInterval is less than the opentelemetry CollectPeriod?  If my understanding is correct, can we document that the CollectPeriod should be at least the ReportingInterval?

**Comment by jsuereth** [comment]
> @dashpole ... the opencensus bridge might skip points if the opencensus ReportingInterval is less than the opentelemetry CollectPeriod?

OpenCensus reports cumulative metrics, so this is probably ok if you're actually exporting a gauge or sum, because the last seens value is the "latest cumulative" so dropping is "free" there.

As an FYI -> I'm trying to migrate the OpenCensus bridge in Java to read directly against the OpenCensus storage rather than use an interval-based reading hook.  Not sure that applies to Go, but look [here for the details](https://github.com/open-telemetry/opentelemetry-java/pull/3835).


+1 to removing Exact distribution @jmacd !! This was very confusing name.

**Comment by jmacd** [comment]
@dashpole @jsuereth If I understand correctly, there is no loss of data. The OTLP exporter supported the Exact aggregator by making an array of gauges. I've replaced that with a bunch of normal Gauges that will be joined into the same array of gauges here: https://github.com/open-telemetry/opentelemetry-go/blob/05c4f109e485f9352fc32e385be7decff03d7b9f/exporters/otlp/otlpmetric/internal/metrictransform/metric.go#L209

(See also #2119)

**Comment by jmacd** [comment]
I should add that #2119 is calling to eliminate the `append()` that joins multiple point values into an array, because it seems like an unnecessary map-building exercise for the ordinary code path, which exports one point per attribute set, thus we expect len(DataPoints) == 1 in the common case.

#2119 also calls to make the reader interface 3-levels from 2-levels, so the export pattern would be like:
```
for each library {
  for each instrument {
    for each attribute set {
      write(point)
    }
  }
}
```

You could make the argument for a 4-level pattern w/ an additional `for each point { ... }` where we support collection interval != export interval. This may be necessary to finish View support in this SDK.

**Review by dashpole — APPROVED** [review]

**Comment by dashpole** [comment]
Thanks for the explanation!

File: CHANGELOG.md [hunk 4]
```diff
@@ -22,7 +22,11 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 - Add the `"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc".WithGRPCConn` option so the exporter can reuse an existing gRPC connection. (#2002)
 - Added a new `schema` module to help parse Schema Files in OTEP 0152 format. (#2267)
-- Added a new `MapCarrier` to the `go.opentelemetry.io/otel/propagation` package to hold propagated coss-cutting concerns as a `map[string]string` held in memory. (#2334)
+- Added a new `MapCarrier` to the `go.opentelemetry.io/otel/propagation` package to hold propagated cross-cutting concerns as a `map[string]string` held in memory. (#2334)
+
+### Removed
+
+- Metric SDK removes the "exact" aggregator for histogram instruments, as it performed a non-standard aggregation for OTLP export (creating repeated Gauge points) and worked its way into a number of confusing examples.
```
Comment (MrAlias) [inline CHANGELOG.md:29]:
```suggestion
- Metric SDK removes the "exact" aggregator for histogram instruments, as it performed a non-standard aggregation for OTLP export (creating repeated Gauge points) and worked its way into a number of confusing examples. (#2348)
```

File: bridge/opencensus/aggregation.go [hunk 5]
```diff
@@ -31,127 +31,111 @@ var (
...
 
-// newAggregationFromPoints creates an OpenTelemetry aggregation from
-// OpenCensus points.  Points may not be empty and must be either
+// recordAggregationsFromPoints records one OpenTelemetry aggregation for
+// each OpenCensus point.  Points may not be empty and must be either
 // all (int|float)64 or all *metricdata.Distribution.
-func newAggregationFromPoints(points []metricdata.Point) (aggregationWithEndTime, error) {
+func recordAggregationsFromPoints(
+	points []metricdata.Point,
+	recorder recordFunc) error {
```
Comment (MrAlias) [inline bridge/opencensus/aggregation.go:41]:
```suggestion
func recordAggregationsFromPoints(points []metricdata.Point, recorder recordFunc) error {
```

**Review by MrAlias — APPROVED** [review]

**Pull request merged**